src/main/java/dev/sobue/slide/controller/TestController.java

```diff
@@ -0,0 +1,40 @@
+package dev.sobue.slide.controller;
+
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Random;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api")
+public class TestController {
+
+  private final Random random = new Random();
+
+  @GetMapping("/test")
+  public ResponseEntity<Void> test() throws Exception {
+    InputStream a1 = getClass().getResourceAsStream("/application.yml");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *a1*. Currently, there are execution paths that do not contain closure statements. Either a) close *a1* in a try-finally block or b) close the resource by declaring *a1* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).